### PR TITLE
Added framework for online docs

### PR DESCRIPTION
**Pull request by RomanMatthew**
address #1 - still need some help

### CODE REVIEW

**Comment by RomanMatthew** [comment]
@CJ-Wright I noticed that the documentation written into the code was of the same style as previous python packages we've released and that there should be some way to easily pull that into the docs, but I think I'm missing something from my conf.py file to allow that to happen. Tips?

**Comment by CJ-Wright** [comment]
You might look at a conf.py here https://github.com/xpdAcq/xpdAn/blob/master/docs/conf.py

**Comment by RomanMatthew** [comment]
@CJ-Wright I'm working on the conf.py file based on the one you presented and hoping it renders the text like hopefully it should. I have to run to a meeting right now but could you review my next commit to see if the `os` stuff is done right? That's what I'm most shaky about

**Comment by RomanMatthew** [comment]
@CJ-Wright It seems like the problem now is the module importation but I really can't figure out what I'm doing wrong. Could you weigh in when you could? Or perhaps @pavoljuhas if you get the chance?

**Comment by CJ-Wright** [comment]
Don't worry about the template stuff (line 96 down) you can remove that.

**Comment by RomanMatthew** [comment]
@CJ-Wright @pavoljuhas this is, I think, my last main problem - I don't quite know how to make the system "find" the documentation written into the python files (which, to my understanding, is where the meat of the docs come from). Can you help me figure it out? 

Screen Shot 2019-07-18 at 1 58 19 PM

**Comment by pavoljuhas** [comment]
> I think, my last main problem - I don't quite know how to make the system "find" the documentation written into the python files

If you use the sphinx `autodoc` extension, the package modules must be importable from the sphinx process (sphinx itself is a Python app).  This can be accomplished by adding to `sys.path` in conf.py as for example [here](https://github.com/diffpy/diffpy.structure/blob/df264a84139bc834424cc323d1bcfc1f9a4ca6cc/doc/manual/source/conf.py#L25) or another option is to run `python setup.py develop` so your source tree is visible to Python.  (assuming `which python` is the same interpreter as the one for sphinx-build)

The first approach is a bit safer, because it will find Python sources from your repo first rather than any pre-existing installation of diffpy.pdfmorph.

**Comment by RomanMatthew** [comment]
@pavoljuhas @sbillinge Do you think you could give it a look now? If you can preview the docs, that is! Got the text up and working and looking mostly good, with a few more things I'd like to add once Simon and I finish hashing out the tutorial!

**Comment by sbillinge** [comment]
@RomanMatthew pls can you send me instructions how to build the docs?

I cloned your branch and I am running ``make html`` in the directory with the Makefile in it but I am getting a moduleNotFound error:  ``ModuleNotFoundError: No module named 'sphinx_rtd_theme'``

What commands are you using to build the docs?

File: doc/source/index.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Finally, we note that though ``PDFmorph`` should work on other spectra 
+that are not PDFs, it has not been extensively tested beyond the PDF.
+
+
+Authors
+-------
+
+``PDFmorph`` is developed by members of the Billinge Group at 
+Columbia University and Brookhaven National Laboratory including 
+Pavol Juhás, Christopher L. Farrow, Christopher Wright, Chia-Hao
```
Comment (sbillinge) [inline doc/source/index.rst]: Chris Farrow first, please fix parens in Tim's name

File: doc/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+#
+import sys
+import os
+sys.path.insert(0, os.path.abspath('../..'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'PDFmorph'
+copyright = '2019, Chris Farrow, Pavol Juhás, Chia-Hao (Timothy) Liu, S. Matthew Roman, Simon J.L. Billinge'
```
Comment (sbillinge) [inline doc/source/conf.py]: copyright is Columbia.  Please get it from the license file.

File: doc/source/index.rst (hunk 1 above)
Comment (CJ-Wright) [inline doc/source/index.rst]:
```suggestion
Pavol Juhás, Christopher L. Farrow, Christopher J. Wright, Chia-Hao 
```

**Pull request merged**